internal/tiltfile/config/config.go

```diff
@@ -25,12 +25,20 @@ type Settings struct {
 	seenWorkingDirectory string
 }
 
+// e.g., "up", "down", "ci"
+type TiltSubcommand string
+
 type Extension struct {
 	UserConfigState model.UserConfigState
+	TiltSubcommand  TiltSubcommand
 }
 
-func NewExtension(userConfigState model.UserConfigState) *Extension {
-	return &Extension{UserConfigState: userConfigState}
+type ExtensionProvider func(userConfigState model.UserConfigState) *Extension
+
+func NewExtensionProvider(tiltSubcommand TiltSubcommand) ExtensionProvider {
```

Comment:
this is ok for now, but this pattern seems like a red flag that this particular extension isn't designed well.

I think the problem is that most extensions have a life-cycle that matches the lifetime of the whole tilt app, but this extensions lifetime is scoped to a single tiltfile execution. We probably need a better way to inject state into the `OnStart` hook

Reply:
yeah, it was already kind of weird that we're putting userConfigState in the extension and then propagating it to the extension state in NewState, and it'd be nice to avoid that as well

anyway, it turns out it was pretty easy to just initialize the field on the constructor right before execution, which still might be a bit ugly but I think is at least 1) more centralized ugliness and 2) closer to where we want to be in the future